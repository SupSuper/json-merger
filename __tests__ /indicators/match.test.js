const helpers = require("../../__helpers__");

describe("when merging two objects and a source property has an $match indicator", function () {

    test("it should use the $match json path result as value", function () {

        const object1 = {
            "a": {
                "aa": "original1",
                "ab": "original2"
            }
        };

        const object2 = {
            "a": {
                "ac": {
                    "$match": "$.ab"
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });
});

describe("when merging two arrays and a source property has an $match indicator and an $remove indicator", function () {

    test("it should remove the item matched by the $match json path", function () {

        const object1 = {
            "a": [
                1,
                2,
                3
            ]
        };

        const object2 = {
            "a": [
                {
                    "$match": "$[1]",
                    "$remove": true
                }
            ]
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });
});
