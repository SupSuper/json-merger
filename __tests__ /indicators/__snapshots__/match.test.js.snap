// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two arrays and a source property has an $match indicator and an $remove indicator it should remove the item matched by the $match json path 1`] = `
"{
	\\"a\\": [
		1,
		3
	]
}"
`;

exports[`when merging two objects and a source property has an $match indicator it should use the $match json path result as value 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original1\\",
		\\"ab\\": \\"original2\\",
		\\"ac\\": {}
	}
}"
`;
